lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -6810,6 +6870,128 @@ Tag('ltx:XMDual', 'afterClose:late' => sub {
+  foreach my $arg (@args) {
+    my $id = LaTeXML::Package::getXMArgID();
+    push(@revargs, Tokens(T_PARAM, T_OTHER(ToString($id))));
+    push(@pargs, Invocation(T_CS('\lx@xmarg'), $id, $arg));
+    push(@cargs, Invocation(T_CS('\lx@xmref'), $id)); }
+  my $optional = undef;
+  if ($keyvals) {
+    my @options = ();
+    foreach my $key (keys %$keyvals) {
+      my $value = $$keyvals{$key};
+      $value = TokenizeInternal($value) if $value && !ref $value;
+      if ($key =~ /^(?:presentation_|content_|)reversion$/) {
+        $value = $value->substituteParameters(@revargs); }
+      push(@options, T_OTHER(',')) if @options;
+      push(@options, T_OTHER($key), T_OTHER('='), T_BEGIN, $value, T_END); }
+    $optional = Tokens(@options); }
+  return
+    Invocation(T_CS('\lx@dual'), $optional,
+    $content->substituteParameters(@cargs),
+    I_wrap({}, $presentation->substituteParameters(@pargs))); }
+
+# A little helper to shorten things up a bit; simply generates #1 (or whatever)
+sub I_arg { return Tokens(T_PARAM, T_OTHER($_[0])); }
+
+sub I_xmarg {
+  my ($id, $arg) = @_;
+  return Tokens(T_CS('\lx@xmarg'),
+    T_BEGIN, (ref $id ? $id : T_OTHER($id)), T_END, T_BEGIN, $arg, T_END); }
+
+sub I_xmref {
+  my ($id) = @_;
+  return Tokens(T_CS('\lx@xmref'), T_BEGIN, (ref $id ? $id : T_OTHER($id)), T_END); }
+
+#----------------------------------------------------------------------
+# This group should be renamed to \lx@somethings and deprecated
+# NOTE: work through this systematically!
+DefMacro('\FCN{}',            '\lx@wrap[role=FUNCTION]{#1}');
+DefMacro('\ROLE{}{}',         '\lx@wrap[role={#1}]{#2}');
+DefMacro('\@SYMBOL{}',        '\lx@wrap[role=ID]{#1}');
+DefMacro('\@CSYMBOL{}',       '\lx@symbol[meaning={#1}]{}');
+DefMacro('\@APPLY{}',         '\lx@apply[]{#1}{}');                             # Sorta broken?
+DefMacro('\@MAYBEAPPLY{}{}',  '\ifx.#2.#1\else\lx@apply{#1}{#2}\fi');
+DefMacro('\@WRAP{}',          '\lx@wrap[]{#1}');
+DefMacro('\@TOKEN{}',         '\lx@symbol[name={#1}]{}');
+DefMacro('\@SUPERSCRIPT{}{}', '\ifx.#2.#1\else\lx@superscript[]{#1}{#2}\fi');
+DefMacro('\@SUBSCRIPT{}{}',   '\ifx.#2.#1\else\lx@subscript[]{#1}{#2}\fi');
+Let('\@PADDED',         '\lx@padded');
+Let('\DUAL',            '\lx@dual');
+Let('\@XMArg',          '\lx@xmarg');
+Let('\@XMRef',          '\lx@xmref');
+Let('\@APPLYFUNCTION',  '\lx@ApplyFunction');
+Let('\@INVISIBLETIMES', '\lx@InvisibleTimes');
+Let('\@INVISIBLECOMMA', '\lx@InvisibleComma');
+Let('\@INVISIBLEPLUS',  '\lx@InvisiblePlus');
+
+# End of stuff to be deprecated.
+#----------------------------------------------------------------------
+
+#======================================================================
+
+# We OUGHT to be able to do this using \llap,\rlap,\hss...
+DefMacro('\lx@tweaked{}{}', '\ifmmode\lx@math@tweaked{#1}{#2}\else\lx@text@tweaked{#1}{#2}\fi');
+DefConstructor('\lx@math@tweaked RequiredKeyVals {}',
+  "<ltx:XMWrap %&GetKeyVals(#1)>#2</ltx:XMWrap>",
```

Comment:
Out of curiosity, shouldn't `<ltx:XMWrap #1>` be made to behave in an identical fashion to `<ltx:XMWrap %&GetKeyVals(#1)>`, when the argument is a KeyVals? I haven't checked what it produces now, I'd assume... something suboptimal?

Reply:
Perhaps so; worth an experiment & possible enhancement. But actually, I'd intended to use `$XMath_attributes` so it only propogaged a known set of attributes.